windows/devices/uwb/UwbConnector.cxx

```diff
@@ -895,43 +868,256 @@ UwbConnector::NotificationListenerStop()
     m_notificationThread.request_stop();
 }
 
-std::weak_ptr<::uwb::RegisteredCallbackToken>
-UwbConnector::RegisterDeviceEventCallbacks(std::weak_ptr<::uwb::UwbRegisteredDeviceEventCallbacks> callbacks)
+/**
+ * @brief Internal helper function that tokenizes a callback if it can be resolved
+ *
+ * @tparam L the type of the callback lambda
+ * @param callbacks the struct holding the callbacks
+ * @param accessor the accessor for the specific callback
+ * @param tokenizer the function that actually tokenizes the callback and gives ownership to the UwbConnector
+ * @return std::weak_ptr<::uwb::RegisteredDeviceCallbackToken>
+ */
+template <typename L>
+std::weak_ptr<::uwb::RegisteredDeviceCallbackToken>
+GetToken(::uwb::UwbRegisteredDeviceEventCallbacks callbacks, std::function<std::weak_ptr<L>(::uwb::UwbRegisteredDeviceEventCallbacks)> accessor, std::function<std::weak_ptr<::uwb::RegisteredDeviceCallbackToken>(std::weak_ptr<L>)> tokenizer)
 {
-    std::lock_guard eventCallbacksLockExclusive{ m_eventCallbacksGate };
-    bool isFirstCallback = not CallbacksPresent();
-    auto token = std::make_shared<::uwb::RegisteredDeviceCallbackToken>(callbacks);
-    m_deviceEventCallbacks.push_back(token);
-    if (isFirstCallback) {
-        NotificationListenerStart();
+    auto callbackWeak = accessor(callbacks);
+    auto callbackShared = callbackWeak.lock();
+    if (not callbackShared) {
+        return std::shared_ptr<::uwb::RegisteredDeviceCallbackToken>();
+    }
+    auto callback = *callbackShared;
+    if (not callback) {
+        return std::shared_ptr<::uwb::RegisteredDeviceCallbackToken>();
     }
-    return token;
+    return tokenizer(callbackWeak);
 }
 
-std::weak_ptr<::uwb::RegisteredCallbackToken>
-UwbConnector::RegisterSessionEventCallbacks(uint32_t sessionId, std::weak_ptr<::uwb::UwbRegisteredSessionEventCallbacks> callbacks)
+::uwb::UwbRegisteredDeviceEventCallbackTokens
+UwbConnector::RegisterDeviceEventCallbacks(::uwb::UwbRegisteredDeviceEventCallbacks callbacks)
 {
     std::lock_guard eventCallbacksLockExclusive{ m_eventCallbacksGate };
     bool isFirstCallback = not CallbacksPresent();
-    auto token = std::make_shared<::uwb::RegisteredSessionCallbackToken>(sessionId, callbacks);
-    auto node = m_sessionEventCallbacks.extract(sessionId);
+
+    auto OnStatusChangedToken = GetToken<::uwb::UwbRegisteredDeviceEventCallbackTypes::OnStatusChanged>(
+        callbacks, [](auto&& cbs) {

```

Comment:
nit: Full word instead of cbs. Obviously can't use "callbacks", but maybe "eventCallbacks" or something similar

Reply:
Did this get updated?

Reply:
oops sorry, I didn't check if I updated it for all of them. I just pushed a change for it: let me know if I missed any.